### PR TITLE
Use rack middleware in sinatra middleware

### PR DESCRIPTION
**Pull request by chrisholmes**
This commit switches the Sinatra instrumentation to run on top of the Rack instrumentation, which is desgined is to be integrated as an extension of other instrumentations.

Extending Rack unlocks some useful configuration options for Sinatra users such as the ability to not trace selected endpoints.

This PR takes a similar approach to the ActionPack instrumentation for integrating with the Rack middleware.

### CODE REVIEW

File: instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb [hunk 1]
```diff
@@ -15,35 +16,15 @@ def initialize(app)
...
-              tracer.in_span(
-                env['PATH_INFO'],
-                attributes: { 'http.method' => env['REQUEST_METHOD'],
-                              'http.url' => env['PATH_INFO'] },
-                kind: :server
-              ) do |span|
-                @app.call(env).tap { |resp| trace_response(span, env, resp) }
-              end
-            end
+            @app.call(env).tap { trace_response(env) }
```
Comment (arielvalentin) [inline instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb:19]: Will this bypass trace response if the request raises an error?

File: instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/instrumentation.rb [hunk 2]
```diff
@@ -13,6 +13,8 @@ module Sinatra
       # instrumentation
       class Instrumentation < OpenTelemetry::Instrumentation::Base
         install do |_|
+          OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.install({})
```
Comment (arielvalentin) [inline instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/instrumentation.rb:16]: The end user still has to manually add the middleware correct? 
Is the Sinatra middleware stack independent of the Rack stack? will this always ensure the middleware is installed after the Rack middleware is installed?

**Review by arielvalentin — COMMENTED** [review]
Are the collector exporter test renames related to this PR?

File: instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/extensions/tracer_extension.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def render(_engine, data, *)
                 end
               end
             end
-
+            app.use OpenTelemetry::Instrumentation::Rack::Middlewares::TracerMiddleware
```
Comment (arielvalentin) [inline instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/extensions/tracer_extension.rb:31]: OOOOOPPPSS I missed this line here

File: instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/instrumentation.rb (hunk 2 above)
Comment (arielvalentin) [reply instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/instrumentation.rb:16]: NVM I missed the injection int eh extensions file.

File: instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb (hunk 1 above)
Comment (chrisholmes) [reply instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb]: good spot, I've made a change for this

**Comment by chrisholmes** [comment]
> Are the collector exporter test renames related to this PR?

yeah, it's not strictly necessary, but it does make my text editor navigation happy. I can delete the commit if you'd prefer it.

**Review by arielvalentin — APPROVED** [review]

**Review by plantfansam — APPROVED** [review]
Stupendous! Thanks for contributing!

File: instrumentation/sinatra/test/opentelemetry/instrumentation/sinatra_test.rb [hunk 4]
```diff
@@ -130,9 +138,24 @@
...
       _(exporter.finished_spans.first.attributes).must_equal(
+        'http.host' => 'example.org',
+        'http.method' => 'GET',
+        'http.scheme' => 'http',
+        'http.status_code' => 404,
+        'http.target' => '/missing_example/not_present'
+      )
+    end
+
+    it 'does correctly name spans when the app raises errors' do
```
[the remark this answers is not in the document]
Comment (chrisholmes) [reply instrumentation/sinatra/test/opentelemetry/instrumentation/sinatra_test.rb:149]: you're right, it should probably test for both and the definition should be updated as such

**Review by plantfansam — APPROVED** [review]
👍 🎉

**Review by robbkidd — APPROVED** [review]
Sinatra approves.

**Comment by plantfansam** [comment]
Hello, and thank you for your contribution!

We recently split Ruby instrumentation out into the [`opentelemetry-ruby-contrib` repo](https://github.com/open-telemetry/opentelemetry-ruby-contrib).

Portions of this PR are related to instrumentation, so we'll need you to re-open it against `opentelemetry-ruby-contrib`. Sorry for the inconvenience! 

To do that, you can:

1. [Create a fork of `opentelemetry-ruby-contrib`](https://github.com/open-telemetry/opentelemetry-ruby-contrib/fork) and copy the git url
1. In your `opentelemetry-ruby` repo, run `git remote add tmp-contrib <your-fork-git-url>`
1. `git push tmp-contrib your-branch-name`
1. Open a new PR in contrib (feel free to just copy/paste your original PR description there)
1. Close your open PR in this repo with a comment that links to your new PR in contrib
1. Delete your `tmp-contrib` remote from `opentelemetry-ruby` (`git remote rm tmp-contrib`)
1. `git clone` your `opentelemetry-ruby-contrib` fork, check out your branch, and make all changes in that repo from now on!

Sorry again for the inconvenience, and thank you for contributing!

**Comment by chrisholmes** [comment]
Moved to the contrib project: https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/54

**Pull request closed without merging**